src/Confluent.Kafka/IConsumer.cs

```diff
@@ -29,171 +29,200 @@ namespace Confluent.Kafka
     ///     value deserialization).
     /// </summary>
-    public interface IConsumer<TKey, TValue> : IClient
+    public interface IConsumer : IClient
     {
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey, TValue}.MemberId" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.MemberId" />
         /// </summary>
         /// <value></value>
         string MemberId { get; }
 
 
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey, TValue}.Consume(TimeSpan)" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.Consume(CancellationToken)" />
         /// </summary>
-        ConsumeResult<TKey, TValue> Consume(TimeSpan timeout);
+        ConsumeResult Consume(CancellationToken cancellationToken = default(CancellationToken));
+
+        
+        /// <summary>
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.Consume(TimeSpan)" />
+        /// </summary>
+        ConsumeResult Consume(TimeSpan timeout);
+
+
+        /// <summary>
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.Consume{TKey, TValue}(CancellationToken)" />
+        /// </summary>
+        ConsumeResult<TKey, TValue> Consume<TKey, TValue>(CancellationToken cancellationToken = default(CancellationToken));
 
 
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey, TValue}.Consume(CancellationToken)" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.Consume{TKey, TValue}(TimeSpan)" />
         /// </summary>
-        ConsumeResult<TKey, TValue> Consume(CancellationToken cancellationToken);
+        ConsumeResult<TKey, TValue> Consume<TKey, TValue>(TimeSpan timeout);
 
 
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey, TValue}.OnPartitionsAssigned" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.OnPartitionsAssigned" />
         /// </summary>
         event EventHandler<List<TopicPartition>> OnPartitionsAssigned;
 
 
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey, TValue}.OnPartitionsRevoked" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.OnPartitionsRevoked" />
         /// </summary>
         event EventHandler<List<TopicPartition>> OnPartitionsRevoked;
 
 
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey, TValue}.OnOffsetsCommitted" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.OnOffsetsCommitted" />
         /// </summary>
         event EventHandler<CommittedOffsets> OnOffsetsCommitted;
 
 
         /// <summary>
-        ///     Refer to <see cref="Confluent.Kafka.Consumer{TKey, TValue}.Assignment" />
+        ///     Refer to <see cref="Confluent.Kafka.Consumer.OnPartitionEOF" />
+        /// </summary>
+        event EventHandler<TopicPartitionOffset> OnPartitionEOF;
```

Comment:
Should the CHANGELOG.md be updated regarding `OnPartitionEOF` the  as part of this pull request? In particular remove
>The Consumer.OnPartitionEOF event has been removed. You should inspect the ConsumeResult instance returned from Consumer.Consume instead.